src/main/java/org/truffleruby/core/fiber/FiberManager.java

```diff
@@ -118,6 +118,8 @@ public Boolean block() throws InterruptedException {
     }
 
     private void fiberMain(RubyContext context, DynamicObject fiber, DynamicObject block, Node currentNode) {
+        assert !Layouts.FIBER.getRootFiber(fiber) : "Root Fibers execute threadMain() and not fiberMain()";
```

Comment:
Not entirely related to this PR, but this is a situation where I really wish this was named `isRootFiber` instead. I had to look up the layout interface to figure out what this was really doing, since I thought it was returning the root fiber itself.

Reply:
Yes, that's unfortunate due the Layout system. We could use a static helper method instead or maybe a default interface method, I'll try.

Reply:
I just removed the field in FiberLayout since it's actually duplicating FiberManager#rootFiber